protocol/p2p/src/core/router.rs

```diff
@@ -19,7 +21,50 @@ use tonic::Streaming;
 
 use super::peer::{PeerKey, PeerProperties};
 
-pub type IncomingRoute = MpscReceiver<KaspadMessage>;
+pub struct IncomingRoute {
+    rx: MpscReceiver<KaspadMessage>,
+    id: u32,
+}
+
+static ROUTE_ID: AtomicU32 = AtomicU32::new(0);
+
+impl IncomingRoute {
+    pub fn new(rx: MpscReceiver<KaspadMessage>) -> Self {
+        let id = ROUTE_ID.fetch_add(1, Ordering::SeqCst);
```

Comment:
bug: `fetch_add` returns the prev value, so the first will be `0` which is considered default and ignored by the router